File: javascript/account.js
const fs = require("fs");
const { randomInt } = require("crypto");
const { parseTXT } = require("./handler.js");

const ACCOUNTS_DB = "./db/accounts.txt";


function _getRandomId(limit = 999999999999) {
    const accounts = parseTXT(ACCOUNTS_DB);
    const usedIds = Object.keys(accounts);
    let generatedId;
    
    var loop = true;
    while (loop) {
        generatedId = String(randomInt(1, limit+1));
        
        if (!usedIds.includes(generatedId)) {
            loop = false;
        }
    }
    
    return generatedId;
}

function getAccount(name, password) {
    const accounts = parseTXT(ACCOUNTS_DB);
    var output = [ null, null ];
    
    for (const accountId in accounts) {
        const account = accounts[accountId];
        
        if (account.name.toLowerCase() === name.toLowerCase() &&
            account.password === password) {
            output[0] = accountId;
            output[1] = account;
            break;
        }
    }
    
    return output;
}

function addAccount(name, phoneNumber, dateBirth, cep, password) {
    const account = getAccount(name, password);
    
    if (account[0] !== null && account[1] !== null) {
        return null;
    }
    
    const generatedId = _getRandomId();
    
    fs.appendFileSync(ACCOUNTS_DB,
        `id           : ${generatedId}\n` +
        `name         : ${name}\n` +
        `phone_number : ${phoneNumber}\n` +
        `date_birth   : ${dateBirth}\n` +
        `cep          : ${cep}\n` +
        `password     : ${password}\n`);
    
    return generatedId;
}


class Account {
    constructor(id, name, phoneNumber, dateBirth, cep) {
        this.id          = id;
        this.name        = name;
        this.phoneNumber = phoneNumber;
        this.dateBirth   = dateBirth;
        this.cep         = cep;
        // this.password    = password;
    }
    
    static login(name, password) {
        const data = getAccount(name, password);
        
        const accountId = data[0]
        const account   = data[1]
        
        if (account === null) {
            return null;
        }
        
        const _name       = account.name
        const phoneNumber = account.phoneNumber;
        const dateBirth   = account.dateBirth;
        const cep         = account.cep;
        
        return new Account(accountId, _name, phoneNumber, dateBirth, cep);
    }
    
    static register(name, phoneNumber, dateBirth, cep, password) {
        const accountId = addAccount(name, phoneNumber, dateBirth, cep, password);
        
        if (accountId === null) {
            return null;
        }
        
        return new Account(accountId, name, phoneNumber, dateBirth, cep);
    }
}

module.exports = {
    ACCOUNTS_DB : ACCOUNTS_DB,
    Account     : Account
};

File: python/account.py
from random import randint
from parser import parse_txt

ACCOUNTS_DB = './db/accounts.txt'


def _get_random_id(limit=999_999_999_999):
    accounts = parse_txt(ACCOUNTS_DB)
    used_ids = []
    generated_id = None
    
    for account_id in accounts:
        used_ids.append(account_id)
    
    while True:
        generated_id = str(randint(1, limit))
        
        if generated_id not in used_ids:
            break
    
    return generated_id

def get_account(name, password):
    accounts = parse_txt(ACCOUNTS_DB)
    
    for account_id in accounts:
        account = accounts[account_id]
        
        if account['name'].lower() == name.lower() and account['password'] == password:
            return account_id, account
    
    return None, None

def add_account(name, phone_number, date_birth, cep, password):
    if get_account(name, password) != (None, None):
        return None
    
    generated_id = _get_random_id()
    
    with open(ACCOUNTS_DB, 'a') as file:
        file.write(
            f'id           : {generated_id}\n'
            f'name         : {name}\n'
            f'phone_number : {phone_number}\n'
            f'date_birth   : {date_birth}\n'
            f'cep          : {cep}\n'
            f'password     : {password}\n'
        )
    
    return generated_id


class Account:
    def __init__(self, _id, name, phone_number, date_birth, cep):
        self.id           = _id
        self.name         = name
        self.phone_number = phone_number
        self.date_birth   = date_birth
        self.cep          = cep
        # self.password     = password
    
    @classmethod
    def login(cls, name, password):
        account_id, account = get_account(name, password)
        
        if account is None:
            return None
        
        _name        = account['name']
        phone_number = account['phone_number']
        date_birth   = account['date_birth']
        cep          = account['cep']
        
        return cls(account_id, _name, phone_number, date_birth, cep)
    
    @classmethod
    def register(cls, name, phone_number, date_birth, cep, password):
        account_id = add_account(name, phone_number, date_birth, cep, password)
        
        if account_id is None:
            return None
        
        return cls(account_id, name, phone_number, date_birth, cep)
